pulp_rpm/tests/functional/conftest.py

```diff
@@ -109,6 +111,64 @@ def signed_artifact(http_get, artifacts_api_client, gen_object_with_cleanup, tmp
     return gen_object_with_cleanup(artifacts_api_client, temp_file).to_dict()
 
 
+@pytest.fixture
+def rpm_artifact_factory(
+    http_get, artifacts_api_client, gen_object_with_cleanup, tmp_path, pulp_domain_enabled
+):
+    """Return an artifact created from uploading an RPM file."""
+
+    def _rpm_artifact_factory(url=RPM_SIGNED_URL, pulp_domain=None):
+        temp_file = tmp_path / str(uuid.uuid4())
+        temp_file.write_bytes(http_get(url))
+        kwargs = {}
+        if pulp_domain:
+            if not pulp_domain_enabled:
+                raise RuntimeError("Server does not have domains enabled.")
+            kwargs["pulp_domain"] = pulp_domain
+        return gen_object_with_cleanup(artifacts_api_client, temp_file, **kwargs).to_dict()
+
+    return _rpm_artifact_factory
+
+
+@pytest.fixture
+def rpm_package_factory(
+    http_get,
+    rpm_package_api,
+    add_to_cleanup,
+    monitor_task,
+    pulp_domain_enabled,
+):
+    """Return a Package created from uploading an RPM file."""
+
+    def _rpm_package_factory(url=RPM_SIGNED_URL, pulp_domain=None):
+        kwargs = {}
+        if pulp_domain:
+            if not pulp_domain_enabled:
+                raise RuntimeError("Server does not have domains enabled.")
+            kwargs["pulp_domain"] = pulp_domain
+
+        with NamedTemporaryFile() as file_to_upload:
+            file_to_upload.write(http_get(url))
```

Comment:
Was the `tmp_path` fixture not working for this one?

Reply:
I was able to take advantage of tmp_path in a number of places. Anywhere NamedTempFile is still in use, it's because dealing with json and PosixPath (eg, json.dump uses write() , have to write_bytes, json.duimps() is a tr, str isn't bytes, bytes needs encoding) ended making the code harder to grok.